src/Raven.Server/Documents/Handlers/Admin/AdminIndexHandler.cs

```diff
@@ -10,6 +11,23 @@ namespace Raven.Server.Documents.Handlers.Admin
 {
     public class AdminIndexHandler : AdminDatabaseRequestHandler
     {
+        [RavenAction("/databases/*/admin/indexes/replace", "POST")]
+        public Task Replace()
+        {
+            //WIP - Efrat
+            var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
+            var index = Database.IndexStore.GetIndex(name);
+            if (index == null)
+                return NoContent();
+            while (true)
+            {
```

Comment:
We give up when we succeed with index replacement (we are waiting for _indexAndTransformerLocker lock)